dfa/src/main/scala/dfa/DFA.scala

```diff
@@ -1,3 +1,19 @@
 package dfa // leave this line in the file
 
-// TODO: replace this comment with your implementation
+case class State(label: String)
+
+case class Transition(from: State, to: State, symbol: Char)
+
+class DFA(val states: Set[State], val transitions: Set[Transition], val start: State, val accept: Set[State]) {
+  def accepts(input: String): Boolean = {
+    var currentState = start
+    for (symbol <- input) {
+      val transitionOpt = transitions.find(t => t.from == currentState && t.symbol == symbol)
```

Comment:
Great use of find here! I used filter to do this but I was left with a set of length 1 so I'd have to access the head of that so this is a much better approach.